pkg/kube2pulumi/kube2pulumi_test.go

```diff
@@ -0,0 +1,333 @@
+	assertion.NoError(err)
+}
+
+func TestSpecialCharPy(t *testing.T) {
+	assertion := assert.New(t)
+
+	_, err := Kube2PulumiFile("../../testdata/specialChar.yaml", "python")
+	assertion.NoError(err)
+}
+
+func TestAnnotationsPy(t *testing.T) {
+	assertion := assert.New(t)
+
+	_, err := Kube2PulumiFile("../../testdata/testDep.yaml", "python")
+	assertion.NoError(err)
+}
+
+func TestMultiLineStringPy(t *testing.T) {
+	assertion := assert.New(t)
+
+	pyExpected, err := ioutil.ReadFile("../../testdata/expectedMultilineString.py")
+	assertion.NoError(err)
+
+	path, err := Kube2PulumiFile("../../testdata/MultilineString.yaml", "python")
+	assertion.NoError(err)
+
+	py, err := ioutil.ReadFile(path)
+	assertion.NoError(err)
+
+	assertion.Equal(string(pyExpected), string(py), "multiline gen is incorrect")
+}
+
+// TYPESCRIPT CODEGEN TESTS
+
+func TestNamespaceTs(t *testing.T) {
+	assertion := assert.New(t)
+
+	tsExpected := `import * as pulumi from "@pulumi/pulumi";
+import * as kubernetes from "@pulumi/kubernetes";
+
+const fooNamespace = new kubernetes.core.v1.Namespace("fooNamespace", {
+    apiVersion: "v1",
+    kind: "Namespace",
+    metadata: {
+        name: "foo",
+    },
+});
+`
+	path, err := Kube2PulumiFile("../../testdata/Namespace.yaml", "typescript")
+	assertion.NoError(err)
+
+	ts, err := ioutil.ReadFile(path)
+	assertion.NoError(err)
+
+	assertion.Equal(tsExpected, string(ts), "typescript codegen is incorrect")
+}
+
+func TestOperatorTs(t *testing.T) {
+	assertion := assert.New(t)
+
+	tsExpected, err := ioutil.ReadFile("../../testdata/k8sOperator/expectedMain.ts")
+	assertion.NoError(err)
+
+	path, err := Kube2PulumiDirectory("../../testdata/k8sOperator/", "typescript")
+	assertion.NoError(err)
+
+	ts, err := ioutil.ReadFile(path)
+	assertion.NoError(err)
+
+	assertion.Equal(string(tsExpected), string(ts), "typescript operator codegen is incorrect")
+}
+
+func TestDoubleQuotesTs(t *testing.T) {
+	assertion := assert.New(t)
+
+	_, err := Kube2PulumiFile("../../testdata/doubleQuotes.yaml", "typescript")
+	assertion.NoError(err)
+}
+
+func TestSpecialCharTs(t *testing.T) {
+	assertion := assert.New(t)
+
+	_, err := Kube2PulumiFile("../../testdata/specialChar.yaml", "typescript")
+	assertion.NoError(err)
+}
+
+func TestAnnotationsTs(t *testing.T) {
+	assertion := assert.New(t)
+
+	_, err := Kube2PulumiFile("../../testdata/testDep.yaml", "typescript")
+	assertion.NoError(err)
+}
+
+func TestMultiLineStringTs(t *testing.T) {
+	assertion := assert.New(t)
+
+	tsExpected, err := ioutil.ReadFile("../../testdata/expectedMultilineString.ts")
+	assertion.NoError(err)
+
+	path, err := Kube2PulumiFile("../../testdata/MultilineString.yaml", "typescript")
+	assertion.NoError(err)
+
+	ts, err := ioutil.ReadFile(path)
+	assertion.NoError(err)
+
+	assertion.Equal(string(tsExpected), string(ts), "multiline gen is incorrect")
+}
+
+// C# CODEGEN TESTS
+
+func TestNamespaceDotNet(t *testing.T) {
```

Comment:
There's a lot of duplicated logic in this file. I think you could simplify this quite a bit by looping through the different languages in each test rather than having a separate test for each language.

As part of that change, it would also make sense to move this inline expected code to a file like you're doing for the operator test. That way, you can reuse more of the logic and just change the language parameter.